VDA-SBT/contracts/SoulboundNFT.sol

```diff
@@ -190,57 +164,58 @@ contract SoulboundNFT is VDAVerificationContract,
 
     }
 
-
     /**
      * @dev See {IsoulboundNFT}
      */
     function claimSBT(
         address did,
-        string calldata sbtType,
-        string calldata uniqueId,
-        string calldata sbtURI,
-        address recipient,
-        bytes calldata signature,
-        bytes calldata proof
+        SBTInfo calldata sbtInfo,
+        bytes calldata requestSignature,
+        bytes calldata requestProof
     ) external override returns(uint) {
-
-        require(isValidSBTType(sbtType), "Invalid SBT type");
+        require(isValidSBTType(sbtInfo.sbtType), "Invalid SBT type");
         {
             bytes memory params = abi.encodePacked(
                 did,
-                "-",
-                sbtType,
-                "-",
-                uniqueId,
-                "-");
-            params = abi.encodePacked(
-                params,
-                sbtURI,
-                "-",
-                recipient,
-                "-");
+                // "-",
+                sbtInfo.sbtType,
+                // "-",
+                sbtInfo.uniqueId,
+                // "-",
+                sbtInfo.sbtURI,
+                // "-",
+                sbtInfo.recipient);
+
             params = abi.encodePacked(
                 params,
-                getNonce(did));
-            verifyRequest(did, _trustedAddresses, params, signature, proof);
+                // "-",
+                sbtInfo.signedData,
+                // "-",
+                sbtInfo.signedProof
+                // "-" 
+            );
+            
+            verifyRequest(did, params, requestSignature, requestProof);
+
+            verifyData(bytes(sbtInfo.uniqueId), sbtInfo.signedData, sbtInfo.signedProof);
```

Comment:
See the original requirements here: https://github.com/verida/blockchain-contracts/issues/57#issuecomment-1298148464

The `proofString` (which you have called `uniqueId`) is in this format:

`twitter-${twitterId}-${did}`

As such, you need to build that proof string in the smart contract:

const proofString = `${sbtType}-${uniqueId}-${did}` and call `verifyData()` on that:

verifyData(proofString, sbtInfo.signedData, sbtInfo.signedProof);

You'll need to update L186 in the test to generate the correct `signedData` on the client side.

Reply:
Let's remove "-" symbols in the proof too as we removed them in the request params.

Reply:
Ok